src/routes/write-exercise/components/WriteExercise.tsx

```diff
@@ -26,230 +50,110 @@ const StyledContainer = styled.View`
-  const [isArticleMissing, setIsArticleMissing] = useState<boolean>(false)
-  // The currently entered answer
-  const [input, setInput] = useState<string>('')
-  // The last submitted answer
-  const [submission, setSubmission] = useState<string | null>(null)
-  // The result of the lastly submitted answer
-  const [result, setResult] = useState<SimpleResultType | null>(null)
-  // The results of all previous documents
-  const [results, setResults] = useState<DocumentResultType[]>([])
-  const [isFocused, setIsFocused] = useState<boolean>(false)
-
-  const touchable: any = React.createRef()
-
-  const document = documents[currentDocumentNumber]
-  const secondAttempt = !!submission
-
-  const current = results.find(result => result.id === document?.id)
-  const nthRetry = current ? current.numberOfTries : 0
-  const needsToBeRepeated = nthRetry < numberOfMaxRetries && (!current || current.result !== SIMPLE_RESULTS.correct)
-
-  const capitalizeFirstLetter = (string: string): string => {
-    return string.charAt(0).toUpperCase() + string.slice(1)
-  }
-
-  const checkEntry = async (): Promise<void> => {
-    const splitInput = input.trim().split(' ')
-
-    if (splitInput.length < 2) {
-      setIsArticleMissing(true)
-      return
-    }
-
-    setSubmission(input)
-    setIsArticleMissing(false)
-
-    const article = capitalizeFirstLetter(splitInput[0])
-    const word = splitInput[1]
-
-    const newResult = validateAnswer(article, word)
-    setResult(newResult)
-
-    storeResult(newResult)
-  }
-
-  const validateAnswer = (article: string, word: string): SimpleResultType => {
-    const validAnswers = [{ article: document.article, word: document.word }, ...document.alternatives]
-    if (validAnswers.some(answer => answer.word === word && answer.article.value === article)) {
-      // Article and word are an exact match with either the document or an alternative
-      return 'correct'
-    } else if (validAnswers.some(answer => answer.word === word)) {
-      // Word is an exact match with either the document or an alternative -> just the article is wrong
-      return 'similar'
-    } else if (
-      validAnswers.some(answer => stringSimilarity.compareTwoStrings(answer.word, word) > almostCorrectThreshold)
-    ) {
-      return 'similar'
-    }
-
-    return 'incorrect'
-  }
+const WriteExercise = ({ documents, route, navigation }: WriteExercisePropType): ReactElement => {
+  const [currentIndex, setCurrentIndex] = useState(0)
+  const [isAnswerSubmitted, setIsAnswerSubmitted] = useState<boolean>(false)
+  const [documentsWithResults, setDocumentsWithResults] = useState<DocumentResultType[]>(
+    documents.map(document => ({
+      ...document,
+      result: null,
+      numberOfTries: 0
+    }))
+  )
 
-  const giveUp = async (): Promise<void> => {
-    const previousResult = result
-    setResult('incorrect')
-    storeResult(previousResult ?? 'incorrect', true)
+  const current = documentsWithResults[currentIndex]
```

Comment:
Checking here that current is define may make sense instead of having to watch out for that in the rest of the component.

Reply:
Actually it isn't even necessary to watch out if current is defined, removed these unnecessary checks.

Reply:
Will be after linting is done :D But you are right :+1: